dbt/include/athena/macros/adapters/metadata.sql

```diff
@@ -1,6 +1,89 @@
 {% macro athena__get_catalog(information_schema, schemas) -%}
-    {{ return(adapter.get_catalog()) }}
-{%- endmacro %}
+  {%- set query -%}
```

Comment:
If you set this, this will not call the adapter function defined in Python or I am missing something ? 🤔

Reply:
From what I understood the adapter function is called first from the docs generate task [here](https://github.com/dbt-labs/dbt-core/blob/45d614533f0c7de81e20488c75c707262c9312e1/core/dbt/task/generate.py#L240), so the order of invocation goes `generate -> adapter.get_catalog -> adapter._get_one_catalog -> get_catalog macro`. So the macro doesn't call the adapter function, is actually the other way around. The only place I found where the `get_catalog` macro is executed was [here](https://github.com/dbt-labs/dbt-core/blob/45d614533f0c7de81e20488c75c707262c9312e1/core/dbt/adapters/base/impl.py#L1069) on the base adapter `_get_one_catalog` function which is overwritten by the Athena adapter anyway.

Reply:
Ok, I check and it seems like that, good to know :)

Reply:
I'm wondered if AWS does not provide possibility to get data catalog details (databases, tables, columns) programmatically for non GLUE catalog types. Just after quick look at their doc I see that CatalogId parameter is required for such API calls, but as I understand we can not get catalog-id ? Did I miss smth ?

Reply:
@maiarareinaldo sorry I resolved too quickly there is comment left

Reply:
hey @svdimchenko, sorry I didn't see your comment before. That's right, for a federated query catalog `get-data-catalog` will return something like this:
```
"DataCatalog": {
    "Name": "catalog_name",
    "Description": "description",
    "Type": "LAMBDA",
    "Parameters": {
        "catalog": "catalog_name",
        "metadata-function": "arn:aws:lambda:<region>:<account_id>:function:<function_name>",
        "record-function": "arn:aws:lambda:<region>:<account_id>:function:<function_name>"
    }
}
```

But you make a very good point, so I looked around and I found this method [list_table_metadata](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena/client/list_table_metadata.html) from Athena client. It returns a pretty similar output to Glue get_tables, the only difference being that that it doesn't have a table "Description" key. I'm replacing the query execution with this method for non Glue catalogs. But I wonder if we should actually use it for Glue catalogs as well, which would removed the need for the if-else when getting the catalog details. Please let me know your thoughts.

Reply:
hi @maiarareinaldo ! I'm so sorry for so long response from my side. Your solution seems good to me.